core/domain/topic_services.py

```diff
@@ -88,6 +89,46 @@ def get_all_skill_ids_assigned_to_some_topic():
     return skill_ids
 
 
+def get_all_skill_ids_assigned_to_some_topic_with_topic_details():
+    """Returns the ids of all the skills that are linked to some topics
+       along with the topic and classroom name to which the
+       topic is assigned to.
+
+    Returns:
+        set([str, str, str]). The ids of all the skills linked to some topic
+            with the topic and classroom name.
+    """
+    skill_ids_with_topic_details = []
+    all_topic_models = topic_models.TopicModel.get_all()
+    all_topics = [
+        topic_fetchers.get_topic_from_model(topic)
+        for topic in all_topic_models]
+
+    for topic in all_topics:
+        for skill_ids in topic.get_all_skill_ids():
+            classroom_name = get_classroom_name_for_topic_with_id(topic.id)
+            obj = skill_ids, topic.name, classroom_name
+            skill_ids_with_topic_details.append(obj)
+    return skill_ids_with_topic_details
+
+
+def get_classroom_name_for_topic_with_id(topic_id):
+    """Return the name of the classroom to which the topic
+    with the given topic id is assigned.
+
+    Returns:
+        str. The name of the classroom assigned to the topic with the given id.
+    """
+    all_classrooms_dict = config_domain.TOPIC_IDS_FOR_CLASSROOM_PAGES.value
+
+    for classroom in all_classrooms_dict:
```

Comment:
Slow -- this is effectively O(N^2) because you have it in another for loop. Build a reverse lookup dict once, to make your query fast, and then read off the values.

Reply:
Done.

Reply:
No, not done. It's still O(N^2) if I'm reading correctly, due to the nested for loop.